packages/grammar-selector/lib/grammar-list-view.js

```diff
@@ -72,32 +88,68 @@ module.exports = class GrammarListView {
   async toggle() {
     if (this.panel != null) {
       this.cancel();
-    } else if (atom.workspace.getActiveTextEditor()) {
-      this.editor = atom.workspace.getActiveTextEditor();
+      return;
+    }
+
+    const editor = atom.workspace.getActiveTextEditor();
+    if (editor) {
+      this.editor = editor;
       this.currentGrammar = this.editor.getGrammar();
       if (this.currentGrammar === atom.grammars.nullGrammar) {
         this.currentGrammar = this.autoDetect;
       }
 
-      const grammars = atom.grammars.getGrammars().filter(grammar => {
+      let grammars = atom.grammars.getGrammars().filter(grammar => {
         return grammar !== atom.grammars.nullGrammar && grammar.name;
       });
+
+      if (atom.config.get('grammar-selector.hideDuplicateTextMateGrammars')) {
```

Comment:
In this case, do you really need the second call to `atom.grammars.getGrammars({ textMateOnly: true }`? It seems like you could simply iterate through the grammars array twice, once to populate the blacklist of Tree Sitter grammars and then a second time to filter. It's not a huge deal though.

Reply:
Yeah, cleaned it up